ext/datadog_profiling_native_extension/heap_recorder.c

```diff
@@ -137,6 +137,13 @@ struct heap_recorder {
   // mutation of the data so iteration can occur without acquiring a lock.
   // NOTE: Contrary to object_records, this table has no ownership of its data.
   st_table *object_records_snapshot;
+  // The GC gen/epoch/count in which we prepared the current iteration.
+  //
+  // This enables us to calculate the age of iterated objects in the above snapshot by
+  // comparing it against an object's alloc_gen.
+  size_t iteration_gen;
+  // The minimum required age for an object to be included in the current iteration.
+  size_t iteration_min_gen_age;
```

Comment:
Minor: It's not a lot of extra code to support this "configurable min age" capability, so I can live with it, but... It does kinda occur to me that this PR would be even smaller if we hardcoded the "1" constant inside the heap recorder instead of making it configurable and hardcoding it in the stack recorder :wink: 

E.g. I wonder if we're ever going to make use of this capability to set `min_age` to something else than 1 :thinking:

Reply:
I was wondering this myself. On the one hand, having the parameter makes the behaviour explicit and gives you an escape hatch to access the full data if needed (e.g. custom debugging/dumping logic). On the other hand, this behavior will be assumed in the UI through our descriptions and what not so any changes may lead to inconsistencies and we wouldn't want to really incentivize this.

I'll hardcode this inside heap for now and we could refer to the intermediate commit with the parameter in the future if we see a need in the future.